src/NServiceBus.Core/Recoverability/DefaultRecoverabilityPolicy.cs

```diff
@@ -18,9 +18,13 @@ public static class DefaultRecoverabilityPolicy
         /// <returns>The recoverability action.</returns>
         public static RecoverabilityAction Invoke(RecoverabilityConfig config, ErrorContext errorContext)
         {
-            if (errorContext.Exception is MessageDeserializationException)
+            // ReSharper disable once LoopCanBeConvertedToQuery
+            foreach (var unrecoverableExceptionType in config.Failed.UnrecoverableExceptionTypes)
```

Comment:
I'm Ok with exact match as well, you propose it for perf reasons right? It would require potentially to configure more exception types but on the other hand it would be easier to understand and document. @SimonCropp any arguments against exact match?